server/fly.io/Cargo.toml

```
@@ -0,0 +1,18 @@
+[package]
+name = "cargo-quickinstall-stats-server"
+version = "0.0.0"
+authors = ["David Laban <alsuren@gmail.com>"]
+# based on https://github.com/fly-apps/hello-rust by Jerome Gravel-Niquet <jeromegn@gmail.com>
+edition = "2021"
+
+# Make a new workspace so that we get our own Cargo.lock and target dir for the docker build. 
+[workspace]
+
+[dependencies]
+tokio = { version = "1", features = ["rt-multi-thread", "macros"] }
+axum = "0.6.10"
+influx_db_client = "^0.5.0"
+influxrs = "2.0.1"
+
```

```suggestion
openssl-sys = { version = "0.9", features = ["vendored"] }

```

this will make sure we use statically linked openssl, when compiling for musl.

---

For the CVEs, we can just setup dependabot to automatically open PR on CVEs and normal dependencies upgrade.

And we could also have an automatic deploy in GHA.

---

We still need to deal with ca-certificates and other nonsense though (unless that gets vendored as well?). 

Can't we just wait for debian-buster to ship the updated openlls+certs and rebuild the docker image?

(I don't know what the mechanism for that would be though, since we're not exactly doing active development on this server, so it won't just happen as part of day-to-day business)

---

> We still need to deal with ca-certificates and other nonsense though (unless that gets vendored as well?). 

Maybe we can try https://github.com/GoogleContainerTools/distroless

> Distroless images are very small. The smallest distroless image, gcr.io/distroless/static-debian12, is around 2 MiB. That's about 50% of the size of alpine (~5 MiB), and less than 2% of the size of debian (124 MiB).

> Can't we just wait for debian-buster to ship the updated openlls+certs and rebuild the docker image?

We won't get any notification for the image, but we can have dependabot automatically opening PR to update rust dependencies and trigger redeployment.

---

I'm going to punt the CVE thing for now (https://github.com/cargo-bins/cargo-quickinstall/issues/277)

re container size: we're still much smaller than anything nodejs or python based, so I'm going to defer this as well. Fly.io hasn't exactly complained, and I've not looked into our request latency numbers, so I don't know if cold starts are likely to be a problem (I assume they wait for us to pass health checks before dumping us in the load balancer, so I doubt it will be a problem).